mmdet/models/detectors/base.py

```diff
@@ -149,4 +150,5 @@ def convert_to_datasample(self, results_list: InstanceList) -> SampleList:
             result = DetDataSample()
             result.pred_instances = results_list[i]
             out_results_list.append(result)
+        samplelist_boxlist2tensor(out_results_list)
```

Comment:
Same question, we should check where converting boxlist to tensor is more reasonable

Reply:
The boxlist results will be converted to tensor at the end of the model.